collie/model/hybrid_matrix_factorization.py

```diff
@@ -322,21 +468,40 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
                 + self.user_biases(users).squeeze(1)
                 + self.item_biases(items).squeeze(1)
             )
-        else:
-            # TODO: remove self.device and let lightning do it
-            metadata_output = self.item_metadata[items, :].to(self.device)
-            if self.metadata_layers is not None:
-                for metadata_nn_layer in self.metadata_layers:
-                    metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(metadata_output)
-                        )
+        elif self.hparams.stage in ('metadata_only', 'all') and self.user_metadata is None:
+            self._calculate_metadata_output(
+                metadata_type='item',
+                users=users,
+                items=items
+            )
+
+            # TODO: make this matrix factorization instead of only a MLP
+            combined_output = torch.cat((self.user_embeddings(users),
```

Comment:
@nathancooperjones not really sure if there is a way to handle the `combined_output` within the ``_calculate_metadata_output`` function

Reply:
I'm actually fine with the way you have it set up here! I think refactoring this to be any less repeated code from this point would actually make it too confusing. How do you feel about this?